### PR TITLE
[BXMSPROD-1881] dependabot for release and blue branches

### PR DESCRIPTION
**Pull request by Ginxo**
related PRs:

* https://github.com/kiegroup/optaplanner-quickstarts/pull/457
* https://github.com/kiegroup/kogito-apps/pull/1512
* https://github.com/kiegroup/kogito-runtimes/pull/2624
* https://github.com/kiegroup/kogito-examples/pull/1485

### CODE REVIEW

**Review by mareknovotny — APPROVED** [review]

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+version: 2
+updates:
+- package-ecosystem: maven
+  directory: "/"
+  schedule:
+    interval: daily
+    time: '03:00'
+  open-pull-requests-limit: 10
+  target-branch: "main"
```
Comment (rsynek) [inline .github/dependabot.yml]: This should have been probably `development`, as there is no `main` branch.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  commit-message:
+    prefix: "[bot][main]"
```
Comment (rsynek) [inline .github/dependabot.yml]: ditto

**Review by rsynek — CHANGES_REQUESTED** [review]
First of all, please open the PR against the `development` branch. The `stable` branch is rather read-only.

As for which branches we cover by `dependabot`, why there is not the `8.29.x`?

**Comment by Ginxo** [comment]
> First of all, please open the PR against the `development` branch. The `stable` branch is rather read-only.
> 
> As for which branches we cover by `dependabot`, why there is not the `8.29.x`?

@rsynek according to documentation dependabot.yml should be placed on default branch from repository, so...
8.29.x branch added

**Comment by rsynek** [comment]
> > First of all, please open the PR against the `development` branch. The `stable` branch is rather read-only.
> > As for which branches we cover by `dependabot`, why there is not the `8.29.x`?
> 
> @rsynek according to documentation dependabot.yml should be placed on default branch from repository, so... 8.29.x branch added

That may well be the case, but every release we reset the `stable` branch to the current release branch. In other words, any changes you make to the `stable` branch directly are only temporary.

**Comment by radtriste** [comment]
> > > First of all, please open the PR against the `development` branch. The `stable` branch is rather read-only.
> > > As for which branches we cover by `dependabot`, why there is not the `8.29.x`?
> > 
> > 
> > @rsynek according to documentation dependabot.yml should be placed on default branch from repository, so... 8.29.x branch added
> 
> That may well be the case, but every release we reset the `stable` branch to the current release branch. In other words, any changes you make to the `stable` branch directly are only temporary.

I agree with @rsynek, we should set it against `development` and not `stable`.
`stable` will be anyway updated in the next release, so no point to update there anything ...

**Comment by mareknovotny** [comment]
you do not merge development to stable branch? @rsynek @radtriste

**Comment by radtriste** [comment]
nope, we recreate the branch directly from the tag

**Comment by Ginxo** [comment]
closed in favor of https://github.com/kiegroup/optaplanner-quickstarts/pull/459

**Pull request closed without merging**